### PR TITLE
Read the VARC transform

### PR DESCRIPTION
**Pull request by rsheeter**
Reading the decomposed transform is relatively useless so I added a method to get affine matrix. That took me on a wee bit of a core_maths adventure.

Builds on #943.

### CODE REVIEW

File: read-fonts/src/tables/varc.rs [hunk 1]
```diff
@@ -152,55 +148,36 @@ impl<'a> VarcComponent<'a> {
...
-        };
-
-        let center_x = if flags.contains(VarcFlags::HAVE_TCENTER_X) {
-            cursor.read::<FWord>()?.to_i16()
-        } else {
-            0
-        };
-        let center_y = if flags.contains(VarcFlags::HAVE_TCENTER_Y) {
-            cursor.read::<FWord>()?.to_i16()
+        let mut transform = DecomposedTransform::default();
```
Comment (cmyr) [inline read-fonts/src/tables/varc.rs:151]: does it make sense to now just have a constructor on this type that takes the cursor and the flags and parses itself?

File: read-fonts/src/tables/varc.rs [hunk 2]
```diff
@@ -215,26 +192,115 @@ impl<'a> VarcComponent<'a> {
...
+            skew_x: 0.0,
+            skew_y: 0.0,
+            center_x: 0.0,
+            center_y: 0.0,
+        }
+    }
+}
+
+impl DecomposedTransform {
+    /// Convert decomposed form to standard 2x3 matrix form
```
Comment (cmyr) [inline read-fonts/src/tables/varc.rs:230]: iirc there is some variation in the world in terms of how the fields of an affine matrix are ordered, and I think it's worth being explicit about the field order in the documentation (see [kurbo, for example](https://docs.rs/kurbo/latest/kurbo/struct.Affine.html#method.new))

**Review by cmyr — APPROVED** [review]
A couple ideas, nothing blocking

File: read-fonts/src/tables/varc.rs (hunk 2 above)
Comment (rsheeter) [reply read-fonts/src/tables/varc.rs:230]: Oh good point. I tend to think of kurbo-style (a b are x-basis, c d are y-basis, e f are translation) as standard but it is not at all true.

File: read-fonts/src/tables/varc.rs (hunk 1 above)
Comment (rsheeter) [reply read-fonts/src/tables/varc.rs:151]: My rationale for the current structure was to keep the bytestream parsing together. I would like to keep that unless you feel strongly about it. If so I will address in a follow-on.

**Pull request merged**